src/Symfony/Component/Serializer/Mapping/Factory/CompiledClassMetadataFactory.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Mapping\Factory;
+
+use Symfony\Component\Serializer\Mapping\AttributeMetadata;
+use Symfony\Component\Serializer\Mapping\ClassDiscriminatorMapping;
+use Symfony\Component\Serializer\Mapping\ClassMetadata;
+
+/**
+ * @author Fabien Bourigault <bourigaultfabien@gmail.com>
+ */
+final class CompiledClassMetadataFactory implements ClassMetadataFactoryInterface
+{
+    private $compiledClassMetadata = [];
+
+    private $loadedClasses = [];
+
+    private $classMetadataFactory;
+
+    public function __construct(string $compiledClassMetadataFile, ClassMetadataFactoryInterface $classMetadataFactory)
+    {
+        if (!file_exists($compiledClassMetadataFile)) {
+            throw new \RuntimeException("File \"{$compiledClassMetadataFile}\" could not be found.");
+        }
+
+        $compiledClassMetadata = require $compiledClassMetadataFile;
+        if (!\is_array($compiledClassMetadata)) {
+            throw new \RuntimeException(sprintf('Compiled metadata must be of the type array, %s given.', \gettype($compiledClassMetadata)));
+        }
+
+        $this->compiledClassMetadata = $compiledClassMetadata;
+        $this->classMetadataFactory = $classMetadataFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getMetadataFor($value)
+    {
+        $className = \is_object($value) ? \get_class($value) : $value;
+
+        if (!isset($this->compiledClassMetadata[$className])) {
+            return $this->classMetadataFactory->getMetadataFor($value);
+        }
+
+        if (!isset($this->loadedClasses[$className])) {
+            $classMetadata = new ClassMetadata($className);
+            foreach ($this->compiledClassMetadata[$className]['attributesMetadata'] as $name => $compiledAttributesMetadata) {
+                $classMetadata->attributesMetadata[$name] = $attributeMetadata = new AttributeMetadata($name);
+                $attributeMetadata->groups = $compiledAttributesMetadata['groups'];
+                $attributeMetadata->maxDepth = $compiledAttributesMetadata['maxDepth'];
+                $attributeMetadata->serializedName = $compiledAttributesMetadata['serializedName'];
```

Comment:
if we remove keys, we can shorten the size of the generated file, and reduce this to 
`[$attributeMetadata->groups, ...] = $compiledAttributesMetadata`

Reply:
I removed all keys (except class name and attribute name). Should I document the format somewhere or just add some `@see` tag somewhere to "link" the `CompiledClassMetadataFactory` to the `ClassMetadataFactoryCompiler`?